worlds/pokemon_emerald/__init__.py

```diff
@@ -848,13 +849,15 @@ def randomize_opponent_parties() -> None:
                     original_species = emerald_data.species[pokemon.species_id]
                     target_bst = sum(original_species.base_stats) if should_match_bst else None
                     target_type = self.random.choice(original_species.types) if should_match_type else None
+                    force_fully_evolved = force_fully_evolved_at != 0 and pokemon.level >= force_fully_evolved_at
 
                     new_species = get_random_species(
                         self.random,
                         self.modified_species,
                         target_bst,
                         target_type,
-                        LEGENDARY_POKEMON if allow_legendaries else set()
+                        LEGENDARY_POKEMON if allow_legendaries else set(),
```

Comment:
Also while porting my patch to the dev branch I saw this which looks reversed? This is a blacklist, which blacklists legendary pokemons when `allow_legendary` is on. Although this is out of scope for this PR, thought I'd let you know :)

Reply:
Not only did I make this mistake here, I made it a second time when refactoring to add this option, lol. So thanks for catching that.